serverless.yaml

```
@@ -16,6 +16,7 @@ custom:
   config: ${file(serverless_config.json)}
   useHapiValidator: ${opt:useHapiValidator, 'false'}
   enableMultiTenancy: ${opt:enableMultiTenancy, 'false'}
+  skipSearchMappingsUpdate: ${opt:skipSearchMappingsUpdate, 'false'}
```

So by default we are turning on the mappings;; could this break any customer?

---

Is it more understandable to say `updateSearchMappings` instead of `skip...` as a paramName?

---

I ended up removing the `skipSearchMappingsUpdate` flag altogether. It was indeed a (small) breaking change. Existing customers could have seen errors on the CFN deployment and would have needed to add the `skipSearchMappingsUpdate` flag.

I ended up modifying the behavior on the search package (https://github.com/awslabs/fhir-works-on-aws-search-es/pull/126) so that it can be configured to ignore errors for the update mappings requests (the `ignoreMappingsErrorsForExistingIndices` param). We'll start off with this param set to true.

This way, all new customers get the new static mappings and it is a non-breaking change for existing customers. Existing customers benefit from this change since any missing index is created with the static mappings, they can check the logs for warnings about mapping conflicts on existing indices and manually reindex with the new mappings, or they ignore them and continue using the already created dynamic mappings (In many cases they work just fine).

---

To confirm will our customers see the warnings about mapping conflicts when they run `sls deploy`? or where would they see the warnings

---

warnings can be seen in the logs of the `updateSearchMappings` Lambda function.

---

Does `disableLogs: true # needed to avoid race condition error "Resource of type 'AWS::Logs::LogGroup' already exists" since the custom resource lambda invocation may create the log group before CFN does` in the CFN stop that from happening

---

In general there's no way to print progress/warning logs on the console when running `sls deploy`.  The exception is when the CFN deployment fails, in that case `sls deploy` prints the failure message.

The `disableLogs: true` is needed for a very peculiar issue with custom resources. Without it the deployment sometimes fails due to "log group with same name already exists error" because serverless attempts to create a log group, but the log group was created seconds before the moment the custom resource started running (the 1st lambda invocation naturally creates a log group )